generators/app/templates/src/app/reducer.ts

```diff
@@ -0,0 +1,48 @@
+import { Nullable } from '~utils/types';
+
+export interface User {
+  id: number;
+}
+
+export interface GlobalState {
+  user: Nullable<User>;
+}
+
+export const INITIAL_STATE = {
+  user: null
+};
+
+enum ActionTypes {
+  SET_USER = 'SET_USER',
+  RESET_USER = 'RESET_USER'
+}
+
+export interface SetUser {
+  type: ActionTypes.SET_USER;
+  payload: User;
+}
+
+export interface ResetUser {
+  type: ActionTypes.RESET_USER;
+}
+
+export type Action = SetUser | ResetUser;
+
+export const actionCreators = {
+  setUser: (user: User) => ({ type: ActionTypes.SET_USER, payload: user }),
+  resetUser: () => ({ type: ActionTypes.RESET_USER })
+};
+
+export const globalReducer = (state: GlobalState, action: Action): GlobalState => {
+  switch (action.type) {
```

Comment:
It's really hard to type 😢

Reply:
I think I told you that in order to have strong action typing I had to use switch. We encountered issues trying to move to a reducerCreator with an object, either because of TS limitations or because we just couldn't figure it out. After a while I decided to go with this as a first approach so as to close this.